eth/tracers/internal/tracetest/util.go

```diff
@@ -17,3 +24,30 @@ func camel(str string) string {
 	}
 	return strings.Join(pieces, "")
 }
+
+type callContext struct {
+	Number     math.HexOrDecimal64   `json:"number"`
+	Difficulty *math.HexOrDecimal256 `json:"difficulty"`
+	Time       math.HexOrDecimal64   `json:"timestamp"`
+	GasLimit   math.HexOrDecimal64   `json:"gasLimit"`
+	Miner      common.Address        `json:"miner"`
+}
+
+func (c *callContext) toBlockContext(genesis *core.Genesis) vm.BlockContext {
+	context := vm.BlockContext{
+		CanTransfer: core.CanTransfer,
+		Transfer:    core.Transfer,
+		Coinbase:    c.Miner,
+		BlockNumber: new(big.Int).SetUint64(uint64(c.Number)),
+		Time:        uint64(c.Time),
+		Difficulty:  (*big.Int)(c.Difficulty),
+		GasLimit:    uint64(c.GasLimit),
+		BaseFee:     genesis.BaseFee,
```

Comment:
I guess base fee also needs to be computed for the current block instead of using the genesis one right? I know this is how it was before. I only realized it now.

Reply:
That sounds reasonable, currently, we're fetching the tx's parent block as the genesis block https://github.com/ethereum/go-ethereum/blob/76490b979ada81e2bbd37380d43a33eb7d087210/eth/tracers/internal/tracetest/makeTest.js#L6-L7

use current block's `BaseFee,ExcessBlobGas,BlobGasUsed` should be more reasonable, what do you think about it?